### PR TITLE
Add perf benchmark utility.

### PR DESCRIPTION
**Pull request by csyonghe**
This allows us to add a `SLANG_PROFILE;` at the beginning of each function to count the number of times it has been called and the total time consumed in the function.

Adds a `-report-perf-benchmark` command line option to `slangc` to dump a diagnostic on benchmark results at the end of compilation.

### CODE REVIEW

File: source/slangc/main.cpp [hunk 1]
```diff
@@ -134,8 +134,8 @@ int wmain(int argc, wchar_t** argv)
         {
             argBuffers.add(args[ii].getBuffer());
         }
-
-        result = MAIN(argc, (char**) &argBuffers[0]);
+        //for (int i = 0; i<10;i++)
+            result = MAIN(argc, (char**) &argBuffers[0]);
```
Comment (expipiplus1) [inline source/slangc/main.cpp:138]: .

**Review by expipiplus1 — APPROVED** [review]

File: source/slangc/main.cpp [hunk 2]
```diff
@@ -134,8 +134,8 @@ int wmain(int argc, wchar_t** argv)
         {
             argBuffers.add(args[ii].getBuffer());
         }
-
-        result = MAIN(argc, (char**) &argBuffers[0]);
+        //for (int i = 0; i<10;i++)
```
Comment (jsmall-nvidia) [inline source/slangc/main.cpp:137]: Left over from testing?

**Review by jsmall-nvidia — APPROVED** [review]
Nice. 

You might want to additionally update the command line docs with 

```
slangc -help-style markdown -h > docs/command-line-slangc-reference.md
```

File: source/slangc/main.cpp (hunk 2 above)
Comment (csyonghe) [reply source/slangc/main.cpp]: Removed.

**Pull request merged**